app/util/api/abstract_clients.py

```diff
@@ -57,23 +64,44 @@ def session(self):
     def base_auth(self):
         return self.user, self.password
 
-    def get(self, url: str, error_msg: str, expected_status_codes: list = None, allow_redirect=False):
-        response = self.session.get(url, auth=self.base_auth, verify=False, timeout=self.requests_timeout,
-                                    allow_redirects=allow_redirect)
+    def get(self, url: str,
+            error_msg: str,
+            expected_status_codes: list = None,
+            allow_redirect: bool = False,
+            headers: dict = None,
+            auth: tuple = None):
+        response = self.session.get(url, verify=False, timeout=self.requests_timeout,
+                                    allow_redirects=allow_redirect, headers=headers if headers else self.headers,
+                                    auth=auth if auth else self.base_auth)
+        self.__verify_response(response, error_msg, expected_status_codes)
+        return response
+
+    def delete(self, url: str, error_msg: str, expected_status_codes: list = None, allow_redirect=False):
```

Comment:
Maybe it would be nice to have one template for all methods, e.g. like in "get" and "post" (method parameters from the new line)